kb-survey/src/app/survey/survey.component.ts

```diff
@@ -37,12 +45,38 @@ export class SurveyComponent implements OnInit {
     });
 
     this.route.queryParams.subscribe((queryParam) => {
-      const { solutionId, ...detailsOfProfile } = queryParam;
-      this.profileDetails = detailsOfProfile;
+      this.profileDetails = queryParam;
     });
-
     this.fetchSurveyDetails();
   }
+  openConfirmationDialog(confirmationParams: any): Promise<boolean> {
+    const dialogRef = this.dialog.open(DialogComponent, {
+      data: {
+        title: confirmationParams?.title,
+        message: confirmationParams?.message,
+        actionBtns: confirmationParams?.actionBtns,
+        btnLeftLabel: confirmationParams?.btnLeftLabel,
+        btnRightLabel: confirmationParams?.btnRightLabel,
+      }
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
+      if (result) {
```

Comment:
instead of usin if-esle here, you can use ternary operator for this, code would be much cleaner